### ISSUE TITLE
gettp with flags

### ISSUE THREAD

**Issue by astrofle** [GreenBankObservatory/dysh#432]
**Describe the bug**
When using `gettp` on a flagged `GBTFITSLoad` the resulting spectrum has zeros in the flagged channels instead of blank values.

**How to Reproduce**
```Python
from dysh.fits.gbtfitsload import GBTFITSLoad
sdf = GBTFITSLoad("/home/sdfits/TRCO_241106_KU/TRCO_241106_KU.raw.vegas/")
sdf.gettp(scan=8, ifnum=0, plnum=0, fdnum=0).timeaverage().plot()
```
Results in:

<img width="536" alt="Screenshot 2024-11-25 at 11 22 21 AM" src="https://github.com/user-attachments/assets/d3c679fe-231c-4e07-a83e-759f6b1d045e">

**Environment**
- Dysh version 0.4.0a
- Python version 3.10
- OS

**Comment by mpound** [comment]
The mask was not passed on in TPScan.total_power().  Fixed with commit [e5c2af77](https://github.com/GreenBankObservatory/dysh/commit/e5c2af77642ba72c2080642cef50d14f0edff7d7)

Here is the updated plot:
<img width="369" alt="dysh432fix" src="https://github.com/user-attachments/assets/da18781c-cac7-4e73-a2c9-7caed3c268c8">

**Comment by mpound** [comment]
fixed
